src/main/java/com/marklogic/kafka/connect/source/XmlRowBatcherSupplier.java

```diff
@@ -0,0 +1,101 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Map;
+
+public class XmlRowBatcherSupplier extends RowBatcherSupplier {
+    protected final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final String TABLE_NS_URI = "http://marklogic.com/table";
+    private final String topic;
+    private final DataMovementManager dataMovementManager;
+    private final Map<String, Object> parsedConfig;
+    private final List<SourceRecord> sourceRecords;
+    private final DocumentBuilderFactory factory;
+
+    XmlRowBatcherSupplier(DataMovementManager dataMovementManager, List<SourceRecord> sourceRecords, Map<String, Object> parsedConfig) {
+        this.dataMovementManager = dataMovementManager;
+        this.sourceRecords = sourceRecords;
+        this.parsedConfig = parsedConfig;
+        this.topic = (String) parsedConfig.get(MarkLogicSourceConfig.TOPIC);
+        factory = DocumentBuilderFactory.newInstance();
+    }
+
+    public RowBatcher<Document> get() {
+        ContentHandle<Document> domHandle = new DOMHandle().withMimetype("application/xml");
+        RowBatcher<Document> rowBatcher =  dataMovementManager.newRowBatcher(domHandle);
+        configureRowBatcher(parsedConfig, rowBatcher);
+        rowBatcher.onSuccess(event -> onSuccessHandler(event, sourceRecords));
+        return rowBatcher;
+    }
+
+    protected void onSuccessHandler(RowBatchSuccessListener.RowBatchResponseEvent<?> event, List<SourceRecord> newSourceRecords) {
+        @SuppressWarnings("unchecked") Document rowsDoc = ((RowBatchSuccessListener.RowBatchResponseEvent<Document>) event).getRowsDoc();
+
+        Element docElement = rowsDoc.getDocumentElement();
+        Node columnsNode = docElement.getElementsByTagNameNS(TABLE_NS_URI, "columns").item(0);
+        NodeList rows = docElement.getElementsByTagNameNS(TABLE_NS_URI, "row");
+
+        for (int i=0; i < rows.getLength(); i++) {
+            try {
+                Document newDocument = convertRowNodeToNewDocument(columnsNode, rows.item(i));
+                SourceRecord newRecord = new SourceRecord(null, null, topic, null, documentToString(newDocument));
+                newSourceRecords.add(newRecord);
+            } catch (Exception ex) {
+                logger.error("Failed to create or add SourceRecord from result row; cause: " + ex.getMessage());
```

Comment:
Just do one log message here - we want to treat this as one "log event"; you can just append the "failed result row" after the exception message.